### PR TITLE
sheepdog: allow to delete sheepdog snapshot

### PR DESCRIPTION
**Pull request by vtolstov**
Signed-off-by: Vasiliy Tolstov v.tolstov@selfip.ru

### CODE REVIEW

File: block/sheepdog.c [hunk 1]
```diff
@@ -2047,15 +2047,18 @@ static bool remove_objects(BDRVSheepdogState *s)
 {
     int fd, i = 0, nr_objs = 0;
     Error *local_err = NULL;
-    bool ret = true;
+    int ret = 0;
+    bool ok = true;
```
Comment (mitake) [inline block/sheepdog.c]: let's rename ok to result. ok is unusal naming

File: block/sheepdog.c [hunk 2]
```diff
@@ -2047,15 +2047,18 @@ static bool remove_objects(BDRVSheepdogState *s)
...
     SheepdogInode *inode = &s->inode;
 
+
```
Comment (mitake) [inline block/sheepdog.c]: needless blank line

File: block/sheepdog.c [hunk 3]
```diff
@@ -2541,8 +2545,66 @@ static int sd_snapshot_delete(BlockDriverState *bs,
...
+                pstrcpy(buf + SD_MAX_VDI_LEN, SD_MAX_VDI_TAG_LEN, snap_tag);
+        }
+
+        ret = find_vdi_name(s, s->name, snap_id, snap_tag, &vdi, true,
+                            &local_err);
+        if (ret) {
+                return ret;
+        }
+
+        SheepdogVdiReq hdr = {
```
Comment (mitake) [inline block/sheepdog.c]: please move this declaration of hdr to top of this function

File: block/sheepdog.c [hunk 4]
```diff
@@ -2541,8 +2545,66 @@ static int sd_snapshot_delete(BlockDriverState *bs,
...
+                .opcode = SD_OP_DEL_VDI,
+                .data_length = wlen,
+                .flags = SD_FLAG_CMD_WRITE,
+                .snapid = snap_id,
+        };
+        SheepdogVdiRsp *rsp = (SheepdogVdiRsp *)&hdr;
```
Comment (mitake) [inline block/sheepdog.c]: move declaration of rsp to top, too

File: block/sheepdog.c [hunk 5]
```diff
@@ -2541,8 +2545,66 @@ static int sd_snapshot_delete(BlockDriverState *bs,
                               const char *name,
                               Error **errp)
 {
-    /* FIXME: Delete specified snapshot id.  */
-    return 0;
+        uint32_t snap_id = 0;
```
Comment (mitake) [inline block/sheepdog.c]: please use 4 space indent for qemu code

**Comment by vtolstov** [comment]
@mitake thanks for review. i'm update pr, can you send you patch and my to qemu-devel?

**Pull request closed without merging**